File: server/router/business/bb_pi_treatment_record.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type BbPiTreatmentRecordRouter struct {
}

// InitBbPiTreatmentRecordRouter 初始化 BbPiTreatmentRecord 路由信息
func (s *BbPiTreatmentRecordRouter) InitBbPiTreatmentRecordRouter(Router *gin.RouterGroup) {
	router := Router.Group("bbPiTreatmentRecord").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("bbPiTreatmentRecord")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.BbPiTreatmentRecordApi
	{
		router.POST("createBbPiTreatmentRecord", apiV1.CreateBbPiTreatmentRecord)   // 新建BbPiTreatmentRecord
		router.DELETE("deleteBbPiTreatmentRecord", apiV1.DeleteBbPiTreatmentRecord) // 删除BbPiTreatmentRecord
		router.DELETE("deleteBbPiTreatmentRecordByIds", apiV1.DeleteBbPiTreatmentRecordByIds) // 批量删除BbPiTreatmentRecord
		router.PUT("updateBbPiTreatmentRecord", apiV1.UpdateBbPiTreatmentRecord)    // 更新BbPiTreatmentRecord
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findBbPiTreatmentRecord", apiV1.FindBbPiTreatmentRecord)        // 根据ID获取BbPiTreatmentRecord
		routerNoRecord.GET("getBbPiTreatmentRecordList", apiV1.GetBbPiTreatmentRecordList)  // 获取BbPiTreatmentRecord列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel BbPiTreatmentRecord列表
	}
}

File: web/src/api/bbPiTreatmentRecord.js
import service from '@/utils/request'

// @Tags BbPiTreatmentRecord
// @Summary 创建BbPiTreatmentRecord
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiTreatmentRecord true "创建BbPiTreatmentRecord"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiTreatmentRecord/createBbPiTreatmentRecord [post]
export const createBbPiTreatmentRecord = (data) => {
  return service({
    url: '/bbPiTreatmentRecord/createBbPiTreatmentRecord',
    method: 'post',
    data
  })
}

// @Tags BbPiTreatmentRecord
// @Summary 删除BbPiTreatmentRecord
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiTreatmentRecord true "删除BbPiTreatmentRecord"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /bbPiTreatmentRecord/deleteBbPiTreatmentRecord [delete]
export const deleteBbPiTreatmentRecord = (data) => {
  return service({
    url: '/bbPiTreatmentRecord/deleteBbPiTreatmentRecord',
    method: 'delete',
    data
  })
}

// @Tags BbPiTreatmentRecord
// @Summary 删除BbPiTreatmentRecord
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除BbPiTreatmentRecord"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /bbPiTreatmentRecord/deleteBbPiTreatmentRecord [delete]
export const deleteBbPiTreatmentRecordByIds = (data) => {
  return service({
    url: '/bbPiTreatmentRecord/deleteBbPiTreatmentRecordByIds',
    method: 'delete',
    data
  })
}

// @Tags BbPiTreatmentRecord
// @Summary 更新BbPiTreatmentRecord
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiTreatmentRecord true "更新BbPiTreatmentRecord"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /bbPiTreatmentRecord/updateBbPiTreatmentRecord [put]
export const updateBbPiTreatmentRecord = (data) => {
  return service({
    url: '/bbPiTreatmentRecord/updateBbPiTreatmentRecord',
    method: 'put',
    data
  })
}

// @Tags BbPiTreatmentRecord
// @Summary 用id查询BbPiTreatmentRecord
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.BbPiTreatmentRecord true "用id查询BbPiTreatmentRecord"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /bbPiTreatmentRecord/findBbPiTreatmentRecord [get]
export const findBbPiTreatmentRecord = (params) => {
  return service({
    url: '/bbPiTreatmentRecord/findBbPiTreatmentRecord',
    method: 'get',
    params
  })
}

// @Tags BbPiTreatmentRecord
// @Summary 分页获取BbPiTreatmentRecord列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取BbPiTreatmentRecord列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiTreatmentRecord/getBbPiTreatmentRecordList [get]
export const getBbPiTreatmentRecordList = (params) => {
  return service({
    url: '/bbPiTreatmentRecord/getBbPiTreatmentRecordList',
    method: 'get',
    params
  })
}


// @Tags BbPiTreatmentRecord
// @Summary 快速编辑
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiTreatmentRecord true "快速编辑"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiTreatmentRecord/quickEdit [post]
export const quickEdit = (data) => {
  return service({
    url: '/bbPiTreatmentRecord/quickEdit',
    method: 'post',
    data
  })
}

// @Tags BbPiTreatmentRecord
// @Summary 导出excelBbPiTreatmentRecord列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页导出excelBbPiTreatmentRecord列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiTreatmentRecord/getBbPiTreatmentRecordexcelList [get]
export const excelList = (params) => {
  return service({
    url: '/bbPiTreatmentRecord/excelList',
    method: 'get',
    params
  })
}